Bottom: 84308fcabb5e5944ea5104dc2f1eb9f8f149c370
Top:    562464f4cda183dbde0842adf2f01cbd57a2b9a4
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 17:16:46 +0800

autofs-5.0.7 - fix fix map entry duplicate offset detection


---

diff --git a/lib/cache.c b/lib/cache.c
index 1300130..2d1c2da 100644
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -659,7 +659,7 @@ int cache_update_offset(struct mapent_cache *mc, const char *mkey, const char *k
 
 	me = cache_lookup_distinct(mc, key);
 	if (me && me->age == age) {
-		if (me->multi == owner)
+		if (me == owner || strcmp(me->key, key) == 0)
 			return CHE_DUPLICATE;
 	}
